pkg/server/server.go

```diff
@@ -633,7 +645,10 @@ func NewServerWithOpts(opts ...OpenFGAServiceV1Option) (*Server, error) {
 		s.listUsersDispatchThrottler = throttler.NewConstantRateThrottler(s.listUsersDispatchThrottlingFrequency, "list_users_dispatch_throttle")
 	}
 
-	s.datastore = storagewrappers.NewCachedOpenFGADatastore(storagewrappers.NewContextWrapper(s.datastore), s.maxAuthorizationModelCacheSize)
+	if !s.datastoreContextPropagation {
+		s.datastore = storagewrappers.NewContextWrapper(s.datastore)
+	}
```

Comment:
```suggestion
	if s.datastoreContextPropagation {
		s.datastore = storagewrappers.NewContextWrapper(s.datastore)
	}
```

if the flag is enabled we want to add the context wrapper